src/gui/torrentcontentmodel.cpp

```diff
@@ -312,6 +312,7 @@ bool TorrentContentModel::setData(const QModelIndex &index, const QVariant &valu
             break;
         case TorrentContentModelItem::COL_PRIO:
             item->setPriority(static_cast<BitTorrent::DownloadPriority>(value.toInt()));
+            emit filteredFilesChanged();
```

Comment:
Priority can be changed from "unfiltered" to "unfiltered" state (e.g. normal to maximum) so this signal shouldn't be emitted.

Reply:
Something like this?
```
--- a/src/gui/torrentcontentmodel.cpp
+++ b/src/gui/torrentcontentmodel.cpp
@@ -311,8 +311,14 @@ bool TorrentContentModel::setData(const QModelIndex &index, const QVariant &valu
             item->setName(value.toString());
             break;
         case TorrentContentModelItem::COL_PRIO:
-            item->setPriority(static_cast<BitTorrent::DownloadPriority>(value.toInt()));
-            emit filteredFilesChanged();
+            {
+                const auto previousPrio = item->priority();
+                item->setPriority(static_cast<BitTorrent::DownloadPriority>(value.toInt()));
+                const auto newPrio = item->priority();
+                if ((newPrio != previousPrio) && ((newPrio == BitTorrent::DownloadPriority::Ignored)
+                        || (previousPrio == BitTorrent::DownloadPriority::Ignored)))
+                    emit filteredFilesChanged();
+            }
             break;
         default:
             return false;
```